### PR TITLE
Added airspeed sensor

### PR DESCRIPTION
**Pull request by ahcorde**
Signed-off-by: Alejandro Hernández Cordero <ahcorde@gmail.com>

# 🎉 New feature

## Summary

Added air speed sensor

Depends on:

 - https://github.com/gazebosim/sdformat/pull/1215
 - https://github.com/gazebosim/gz-msgs/pull/321
 - https://github.com/gazebosim/gz-sensors/pull/305

## Checklist
- [ ] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Comment by bperseghetti** [comment]
What's the timeline of this being updated from a draft, would love to start using this upstream!

**Comment by ahcorde** [comment]
@osrf-jenkins run tests please!

**Comment by ahcorde** [comment]
@jennuine and @mjcarroll friendly ping, I would love to merge this, there is a PR in PX4 waiting for a while

**Comment by mjcarroll** [comment]
It looks good, it looks like there aren't tests to exercise some of the new functionality (at least according to coverage)  can you add some?

**Comment by ahcorde** [comment]
@mjcarroll Added test https://github.com/gazebosim/gz-sim/pull/1847/commits/28135bb24077699412aed70a5a180704e86bb09d

**Review by mjcarroll — APPROVED** [review]

File: src/Util.cc [hunk 1]
```diff
@@ -81,6 +83,57 @@ math::Pose3d worldPose(const Entity &_entity,
...
+  // {
+  //   // get pose of parent entity
+  //   auto parentVel = _ecm.Component<components::Pose>(p->Data());
+  //   if (!parentVel)
+  //     break;
+  //   // transform pose
+  //   vel = parentVel->Data() * vel;
+  //   // keep going up the tree
+  //   v = _ecm.Component<components::ParentEntity>(v->Data());
+  // }
```
Comment (iche033) [inline src/Util.cc:133]: remove?

File: src/Util.cc [hunk 2]
```diff
@@ -81,6 +83,57 @@ math::Pose3d worldPose(const Entity &_entity,
...
+  auto worldLinVel = _ecm.Component<components::WorldLinearVelocity>(_entity);
+  if (nullptr == worldLinVel)
+  {
+    gzwarn << "Trying to get world velocity from entity [" << _entity
+            << "], which doesn't have a velocity component" << std::endl;
+    return math::Vector3d();
+  }
+
+  math::Vector3d vel = worldLinVel->Data();
+  auto v = _ecm.Component<components::ParentEntity>(_entity);
```
Comment (iche033) [inline src/Util.cc:122]: unused variable `v`

File: examples/worlds/sensors.sdf [hunk 3]
```diff
@@ -136,21 +136,20 @@
             </vertical_velocity>
           </altimeter>
         </sensor>
-        <sensor name="air_pressure" type="air_pressure">
+        <sensor name="air_speed" type="air_speed">
```
Comment (iche033) [inline examples/worlds/sensors.sdf:139]: should we add a new air_speed sensor (and keep the existing air_pressure sensor)?

File: test/integration/air_speed_system.cc [hunk 4]
```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (C) 2019 Open Source Robotics Foundation
```
Comment (iche033) [inline test/integration/air_speed_system.cc:2]: 2019 - > 2023

File: examples/worlds/sensors.sdf (hunk 3 above)
Comment (iche033) [reply examples/worlds/sensors.sdf:139]: Do you also need to add the `gz-sim-air-speed-system`?

**Pull request merged**